Applications/Spire/Source/Ui/EditableTableView.cpp

```diff
@@ -262,6 +274,32 @@ EditableTableView::EditableTableView(
   get_header().get_item(0)->set_is_resizeable(false);
   get_header().get_widths()->set(0, scale_width(26));
   set_style(*this, TABLE_VIEW_STYLE());
+  setFocusProxy(&get_body());
+  get_body().installEventFilter(this);
+  m_current_connection = get_current()->connect_update_signal(
+    std::bind_front(&EditableTableView::on_current, this));
+}
+
+bool EditableTableView::eventFilter(QObject* watched, QEvent* event) {
+  if(watched == &get_body()) {
+    if(event->type() == QEvent::KeyPress) {
+      auto& key_event = *static_cast<QKeyEvent*>(event);
+      if(key_event.key() == Qt::Key_Shift) {
+        return true;
+      }
+    }
+  }
+  return TableView::eventFilter(watched, event);
+}
+
+bool EditableTableView::focusNextPrevChild(bool next) {
+  setFocus();
```

Comment:
Why does the table set focus on itself? My understanding from Qt's documentation is that this function is supposed to set focus on the next or previous child in the focus chain.

Reply:
focusNextPrevChild itself doesn't change focus.